docs/reference/modules/discovery/discovery.asciidoc

```diff
@@ -0,0 +1,187 @@
+connect to when those IP addresses may not be known at node startup.
+
+To enable file-based discovery, configure the `file` hosts provider as follows:
+
+[source,txt]
+----------------------------------------------------------------
+discovery.zen.hosts_provider: file
+----------------------------------------------------------------
+
+Then create a file at `$ES_PATH_CONF/unicast_hosts.txt` in the format described
+below. Any time a change is made to the `unicast_hosts.txt` file the new changes
+will be picked up by Elasticsearch and the new hosts list will be used.
+
+Note that the file-based discovery plugin augments the unicast hosts list in
+`elasticsearch.yml`: if there are valid unicast host entries in
+`discovery.zen.ping.unicast.hosts` then they will be used in addition to those
+supplied in `unicast_hosts.txt`.
+
+The `discovery.zen.ping.unicast.hosts.resolve_timeout` setting also applies to
+DNS lookups for nodes specified by address via file-based discovery. This is
+specified as a <<time-units, time unit>> and defaults to 5s.
+
+The format of the file is to specify one node entry per line.  Each node entry
+consists of the host (host name or IP address) and an optional transport port
+number.  If the port number is specified, is must come immediately after the
+host (on the same line) separated by a `:`.  If the port number is not
+specified, a default value of 9300 is used.
+
+For example, this is an example of `unicast_hosts.txt` for a cluster with four
+nodes that participate in unicast discovery, some of which are not running on
+the default port:
+
+[source,txt]
+----------------------------------------------------------------
+10.10.10.5
+10.10.10.6:9305
+10.10.10.5:10005
+# an IPv6 address
+[2001:0db8:85a3:0000:0000:8a2e:0370:7334]:9301
+----------------------------------------------------------------
+
+Host names are allowed instead of IP addresses (similar to
+`discovery.zen.ping.unicast.hosts`), and IPv6 addresses must be specified in
+brackets with the port coming after the brackets.
+
+It is also possible to add comments to this file. All comments must appear on
+their lines starting with `#` (i.e. comments cannot start in the middle of a
+line).
+
+[float]
+[[ec2-hosts-provider]]
+===== EC2 hosts provider
+
+The {plugins}/discovery-ec2.html[EC2 discovery plugin] adds a hosts provider
+that uses the https://github.com/aws/aws-sdk-java[AWS API] to find a list of
+seed nodes.
+
+[float]
+[[azure-classic-hosts-provider]]
+===== Azure Classic hosts provider
+
+The {plugins}/discovery-azure-classic.html[Azure Classic discovery plugin] adds
+a hosts provider that uses the Azure Classic API find a list of seed nodes.
+
+[float]
+[[gce-hosts-provider]]
+===== Google Compute Engine hosts provider
+
+The {plugins}/discovery-gce.html[GCE discovery plugin] adds a hosts provider
+that uses the GCE API find a list of seed nodes.
+
+[float]
+==== Discovery settings
```

Comment:
Does this replace https://www.elastic.co/guide/en/elasticsearch/reference/master/discovery-settings.html ? If not, there should be links between the two.

Reply:
Not really, the document you linked is under "important settings" and contains the two important settings that all users will need to interact with. The settings here should almost always be left with their default values.